controllers/openshift_controller.go

```diff
@@ -2426,6 +2449,66 @@ func (r *KataConfigOpenShiftReconciler) createMachineConfigsFromFiles(yamlFiles
 	return nil
 }
 
+// Delete confidential container runtime classes and their MachineConfigs
+func (r *KataConfigOpenShiftReconciler) deleteConfidentialContainerRuntimeClasses() error {
+	// Delete runtime classes
+	runtimeClasses := []string{kataCCIntelRuntimeClassName, kataCCAMDRuntimeClassName}
+
+	for _, runtimeClassName := range runtimeClasses {
+		rc := &nodeapi.RuntimeClass{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "node.k8s.io/v1",
+				Kind:       "RuntimeClass",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: runtimeClassName,
+			},
+		}
+
+		err := r.Client.Delete(context.TODO(), rc)
```

Comment:
This isn't needed because `createRuntimeClass` does :

```
	// Set Kataconfig r.kataConfig as the owner and controller
	if err := controllerutil.SetControllerReference(r.kataConfig, rc, r.Scheme); err != nil {
		return err
	}
```

This causes the RuntimeClass object to be automatically deleted when KataConfig goes away.